core/src/main/java/org/apache/iceberg/BaseDistributedDataScan.java

```diff
@@ -368,7 +369,9 @@ private CloseableIterable<ScanTask> toFileTasks(
           ScanMetricsUtil.fileTask(scanMetrics(), dataFile, deleteFiles);
 
           return new BaseFileScanTask(
-              copyDataFiles ? dataFile.copy(shouldReturnColumnStats()) : dataFile,
+              copyDataFiles
```

Comment:
Optional: I am not a big fan of using ternary operators when they are split across multiple lines. To me, this makes the logic harder to follow. I'd consider adding a method like below to `BaseScan`.

```
protected <F extends ContentFile<F>> F copy(F file) {
  return ContentFileUtil.copy(file, shouldReturnColumnStats(), columnsToKeepStats());
}
```

This will make the invocation shorter and we may reuse this method in other scans in the future.

Reply:
Created the method